cmd/slsa-github-generator/generate.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"io/ioutil"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/spf13/cobra"
+
+	"github.com/slsa-framework/slsa-github-generator/github"
+	"github.com/slsa-framework/slsa-github-generator/signing/sigstore"
+	"github.com/slsa-framework/slsa-github-generator/slsa"
+)
+
+// GenerateCmd returns the 'generate' command.
+func generateCmd() *cobra.Command {
+	var attPath string
+
+	c := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate SLSA provenance from a Github Action",
+		Long: `Generate signed SLSA provenance from a Github Action and upload to a Rekor
+transparency log. This command assumes that it is being run in the context of
+a Github Actions workflow.`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			ghContext, err := github.GetWorkflowContext()
+			check(err)
+
+			p, err := slsa.HostedActionsProvenance(slsa.WorkflowRun{
+				// TODO: Get subject names and digests
+				Subjects:      []intoto.Subject{},
+				BuildType:     "",
```

Comment:
@MarkLodato suggested we use https://github.com/slsa-framework/slsa-github-generator-go/blob/main/pkg/provenance.go#L135. I think this needs set be set dynamically for each ecosystem-specific builder.

Reply:
Yeah, I wasn't sure what to do with this yet but it looks like we'll need it even if we don't set a buildConfig.

Reply:
Added a similar build type

Reply:
The API allows you to specify it already in the WorkflowRun but I'll add a CLI option also.